src/doc/telio_events.md

```diff
@@ -0,0 +1,150 @@
+# Documentation For Describing Telio Events
+
+This module documents the events emitted by the Telio library.
+
+## Telio Error Event
+
+Once registered using telio_new for events. One of the possible reported event is of type [Error][telio_model::event::Error], in format:
+
+```json
+{
+  "type": "error",
+  "body": {
+    "level": "<critical|severe|warning|notice>",
+    "code": "<noerror|unknown>",
+    "msg": string?
+  }
+}
+```
+
+Fields:
+
+- level - error severity:
+
+    - critical - telio panicked
+    - severe - telio can continue operating, but parts of it can malfunction
+    - warning - warning that things can get wrong
+    - notice - not a critical error
+
+- code - common error code:
+
+    - noerror - default value, not an error.
+    - unknown - unknown error
+
+- msg - custom error string, can be empty
+
+
+### When error event is reported?
+
+One of the situations, when error event will be reported, is when panic happens on telio side.
```

Comment:
Hmm, what about other situations?

Reply:
Currently it's the only case there error is reported via handler.

Other minnor ones are reported by ffi it self